trinity/plugins/builtin/attach/plugin.py

```diff
@@ -0,0 +1,65 @@
+from abc import (
+    abstractmethod,
+)
+from argparse import (
+    ArgumentParser,
+    Namespace,
+)
+import sys
+
+from trinity.config import (
+    ChainConfig,
+)
+from trinity.extensibility import (
+    BasePlugin,
+)
+
+from trinity.plugins.builtin.attach.console import (
+    console,
+)
+
+
+class AttachPlugin(BasePlugin):
+
+    @property
+    def name(self) -> str:
+        return "Attach"
+
+    @property
+    @abstractmethod
+    def use_ipython(self) -> bool:
+        raise NotImplementedError("Must be implemented in subclasses")
+
+    def configure_parser(self, arg_parser: ArgumentParser) -> None:
+
+        subparser = arg_parser.add_subparsers(dest='subcommand')
+        attach_parser = subparser.add_parser(
+            'attach',
+            help='open an REPL attached to a currently running chain',
+        )
+
+        attach_parser.set_defaults(func=self.run_console)
+
+    def run_console(self, args: Namespace, chain_config: ChainConfig) -> None:
+        try:
+            console(chain_config.jsonrpc_ipc_path, use_ipython=self.use_ipython)
+        # FIXME: That doesn't seem to work very well as we leave behind the
+        # JSON IPC file by default it seems. Maybe we can make this more reliable
+        # with the upcoming PidFile support.
```

Comment:
I think this is just `IPCServer` leaving behind the file, no matter how it's started. You can remove this comment, I'll post a fixup PR.